theano/sandbox/cuda/corr_gemm.cu

```diff
@@ -54,6 +54,7 @@ inline int GET_BLOCKS(const int N) {
 // Kernels for fast unfold + copy
 __global__ void im2col_kernel(const int n, const float* data_im,
     const int height, const int width, const int kernel_h, const int kernel_w,
+    const int dilation_h, const int dilation_w,
```

Comment:
> But looking at Caffe code, there is high analogy between both codes.

Yes, because I originally copied it from Caffe. Everything that has changed in between is probably because it has changed in Caffe.

> Switched to Caffe implementation both im2col and col2im CUDA kernels.

Cool! I think this is easier to maintain for the future.